pkg/openstack/galera.go

```diff
@@ -28,14 +32,64 @@ const (
 	galeraReady    galeraStatus = iota
 )
 
+func findAndDeleteGalera(
+	ctx context.Context,
+	instance *corev1beta1.OpenStackControlPlane,
+	helper *helper.Helper,
+) (ctrl.Result, error) {
+
+	log := GetLogger(ctx)
+	// Fetch the list of Galera objects
+	galeraList := &mariadbv1.GaleraList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(instance.GetNamespace()),
+	}
+	err := helper.GetClient().List(ctx, galeraList, listOpts...)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("could not get galeras %w", err)
+	}
+
+	var delErrs []error
+	for _, galeraObj := range galeraList.Items {
+		// if not in templateGalera-CR, delete from DB
```

Comment:
nit:
// if it is not defined in  the OpenStackControlPlane then delete it from k8s.